hr_timesheet_sheet_worked_time/models/hr_timesheet_sheet_sheet.py

```diff
@@ -0,0 +1,95 @@
+# -*- coding: utf-8 -*-
+# Copyright 2020 Quartile Limited
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from datetime import datetime, timedelta
+from odoo import models, fields, api
+from odoo.tools import DEFAULT_SERVER_DATE_FORMAT
+
+
+class HrTimesheetSheet(models.Model):
+    _inherit = 'hr_timesheet_sheet.sheet'
+
+    standard_work_hours = fields.Float(
+        compute='_compute_standard_work_hours',
+        string="Standard Work Hours",
+    )
+    expected_work_hours = fields.Float(
+        compute='_compute_expected_work_hours',
+        string="Expected Work Hours",
+    )
+    overtime_hours = fields.Float(
+        compute='_compute_overtime_hours',
+        string="Overtime Hours",
+    )
+    holiday_hours = fields.Float(
+        compute='_compute_holiday_hours',
+        string="Holiday Hours",
+    )
+
+    @api.depends(
+        'employee_id.calendar_id',
+        'employee_id.calendar_id.attendance_ids',
+        'employee_id.calendar_id.attendance_ids.dayofweek'
+    )
+    def _compute_holiday_hours(self):
+        for sheet in self:
+            today_date = fields.Datetime.to_string(datetime.now().date())
+            date_to = sheet.date_to
+            if sheet.date_to > today_date:
+                date_to = today_date
+            public_holiday_line_ids = self.env[
+                'hr.holidays.public.line'].search(
+                [
+                    ('date', '>', self.date_from),
+                    ('date', '<', date_to),
+                    ('year_id.year', '=', datetime.now().year),
+                ])
+            holiday_hours = 0.0
+            for line in public_holiday_line_ids:
+                public_holiday = datetime.strptime(
+                    str(line.date), DEFAULT_SERVER_DATE_FORMAT)
+                attendance_ids = sheet.employee_id.calendar_id.attendance_ids.filtered(
+                    lambda attendance: int(
+                        attendance.dayofweek) == public_holiday.weekday())
+                for attendance in attendance_ids:
+                    holiday_hours += \
+                        attendance.hour_from - attendance.hour_to
+            sheet.holiday_hours = abs(holiday_hours)
```

Comment:
Why we need to take `abs()` here? Is that because the following computation will return negative?
```
holiday_hours += attendance.hour_from - attendance.hour_to
```

Reply:
yes there is a possibility in `attendance.hour_from` will be less value than `attendance.hour_to` so that type of occurrence to avoid I have put abs